nengo/tests/test_dists.py

```diff
@@ -295,3 +302,57 @@ def test_frozen():
     assert hash(a) != hash(c)  # not guaranteed, but highly likely
     assert b != c
     assert hash(b) != hash(c)  # not guaranteed, but highly likely
+
+
+@pytest.mark.parametrize("dist", all_dists)
+def test_dist_calls_super(dist):
+    """Ensures that distributions call the superclass's __init__ method."""
+
+    # This test automatically attempts to create an instance of the object.
+    # It's a pretty tough piece of code to get right in all cases, so this
+    # is only as complicated as is necessary for the actually existing
+    # distributions.
+
+    # If you add a new distribution that fails this test because it cannot
+    # be instantiated, edit this logic so that your distribution works!
+
+    argspec = inspect.getargspec(dist.__init__)
+    args = {arg: False for arg in argspec.args if arg != 'self'}
+
+    inst = None
+    fixes = 0
+
+    while inst is None:
+        try:
+            # Instantiate the distribution
+            inst = dist(**args)
+        except ValidationError as e:
+            # If we can't, we'll apply some fix
+            if fixes > 50:
+                # If we've already tried a lot of fixes, fail
+                raise
+
+            # First, figure out what arg needs fixing
+            msg = str(e)
+            for arg in args:
+                arg_str = "%s.%s" % (dist.__name__, arg)
+                if arg_str in msg:
+                    to_change = arg
+                    break
+            else:
+                # If the error message doesn't include the arg, choose randomly
+                to_change = random.choice(list(args))
+
+            # Try some different values, or fail if we've tried them all
+            if isinstance(args[to_change], bool):
+                args[to_change] = 10
+            elif isinstance(args[to_change], int):
+                args[to_change] = 0.5
+            elif isinstance(args[to_change], float):
+                args[to_change] = np.ones(1)
+            else:
+                print(to_change, arg_str, args)
+                raise
+            fixes += 1
+
+    assert hasattr(inst, '_paramdict')  # Created by FrozenObject.__init__
```

Comment:
The only downside that I can see with testing for implementation details (since those implementation details are contained in this package) is that the test might have to change if the `FrozenDict` implementation changes. That doesn't seem like a big deal to me... anyhow, if the pylint stuff works then this will be moot anyhow, so I'll look into that first!